### PR TITLE
Update hydrobasin path

### PR DESCRIPTION
**Pull request by GbotemiB**
# Closes # (if applicable).

## Changes proposed in this Pull Request
This PR addresss path for hydrobasin which is not set to have base_dir. This leads to problem here when running from outside the root of pypsa-earth with snakemake module.
https://github.com/pypsa-meets-earth/pypsa-earth/blob/main/scripts/build_renewable_profiles.py#L598-L599

https://github.com/pypsa-meets-earth/pypsa-earth/blob/main/scripts/build_renewable_profiles.py#L713

## Checklist

- [x] I consent to the release of this PR's code under the AGPLv3 license and non-code contributions under CC0-1.0 and CC-BY-4.0.
- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [ ] Newly introduced dependencies are added to `envs/environment.yaml` and `doc/requirements.txt`.
- [ ] Changes in configuration options are added in all of `config.default.yaml` and `config.tutorial.yaml`.
- [ ] Add a test config or line additions to `test/` (note tests are changing the config.tutorial.yaml)
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [ ] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

**Comment by yerbol-akhmetov** [comment]
Thanks @GbotemiB for noticing this behavior and fixing it. Basically, Emmanuel found that `resource` which contains data path are later used. That was causing issue when running `build_renewable_profile` rule outside of pypsa-earth directory for hydro profiles. I have tested this implementation locally and it solves the issue. @davide-f

File: scripts/build_renewable_profiles.py [hunk 1]
```diff
@@ -538,7 +538,9 @@ def create_scaling_factor(
     # filter plants for hydro
     if snakemake.wildcards.technology.startswith("hydro"):
         country_shapes = gpd.read_file(paths.country_shapes)
-        hydrobasins = gpd.read_file(os.path.join(BASE_DIR, resource["hydrobasins"]))
+        hydrobasins_path = os.path.join(BASE_DIR, resource["hydrobasins"])
+        resource["hydrobasins"] = hydrobasins_path
```
Comment (ekatef) [inline scripts/build_renewable_profiles.py:542]: Just to be sure: a path to hydrobasins is used further when calculating the inflow [here](https://github.com/pypsa-meets-earth/pypsa-earth/blob/d727d39e95a04a1c1fe7d5935921ad060ad3fdfc/scripts/build_renewable_profiles.py#L594-L596), right?

**Review by ekatef — COMMENTED** [review]
Hello @GbotemiB and @yerbol-akhmetov! Thanks for the fix 😄 
Have added a little comment which is basically a clarification question.

That it's essentially an additional fix to #1137, right? I understand you have tested it and it's working?

File: scripts/build_renewable_profiles.py (hunk 1 above)
Comment (GbotemiB) [reply scripts/build_renewable_profiles.py:542]: Thats right.

**Comment by GbotemiB** [comment]
> Hello @GbotemiB and @yerbol-akhmetov! Thanks for the fix 😄 Have added a little comment which is basically a clarification question.
> 
> That it's essentially an additional fix to #1137, right? I understand you have tested it and it's working?

Yeah. Thats correct. I have tested it and it is working properly.

**Comment by ekatef** [comment]
@GbotemiB perfect, thank you! Merging 😄

**Pull request merged**